libparsers/src/xml/stations_and_request_stops.cpp

```diff
@@ -0,0 +1,430 @@
+#include "parsers/xml/stations_and_request_stops.hpp"
+#include "utils.hpp"
+#include <gsl/gsl_util>
+#include <rapidxml_ns.hpp>
+#include <sstream>
+#include <string>
+#include <unordered_map>
+
+using namespace std::string_literals;
+
+namespace parsers {
+namespace xml {
+	namespace stations_and_stops {
+		namespace {
+			station_marker::doors parse_doors_text(std::string const& filename,
+			                                       rapidxml_ns::xml_node<char>* doors_node,
+			                                       errors::multi_error_t& errors) {
+				static std::unordered_map<std::string, station_marker::doors> text_mapping{
+				    //
+				    {"left", station_marker::doors::left},
+				    {"right", station_marker::doors::right},
+				    {"none", station_marker::doors::none}
+				    //
+				};
+				std::string door;
+				try {
+					door = doors_node->value();
+					util::lower(door);
+				}
+				catch (std::exception const& e) {
+					add_error(errors, filename, 0, e.what());
+					door = "none";
+				}
+				auto const parsed_doors = text_mapping.find(door);
+
+				if (parsed_doors != text_mapping.end())
+					return parsed_doors->second;
+
+				add_error(errors, filename, 0, "Error: <Doors> given and invalid option");
+				return station_marker::doors::none;
+			}
+			request_stop_marker::behaviour parse_ai_behaviour(
+			    std::string const& filename,
+			    rapidxml_ns::xml_node<char>* ai_behaviour_node,
+			    errors::multi_error_t& errors) {
+				static std::unordered_map<std::string, request_stop_marker::behaviour>
+				    text_mapping{{"fullspeed", request_stop_marker::behaviour::fullspeed},
```

Comment:
Use comments at the end of the line to force clang-format to leave your formatting alone.